### PR TITLE
Added initial support for webhooks. Need next version of discord.js b…

### PR DESCRIPTION
**Pull request by ollieh**
…efore file attachments will work through webhooks.

### CODE REVIEW

**Comment by ollieh** [comment]
Just realised if you have multiple rooms on the same discord server that you need to bridge, it won't work as it doesn't filter the hooks by room. I'll fix that.

File: src/bot.ts [hunk 1]
```diff
@@ -168,12 +172,21 @@ export class DiscordBot {
...
+    const webhooks = await chan.fetchWebhooks();
+    const hook : Discord.Webhook = webhooks.filterArray((h) => h.name === "_matrix").pop();
     try {
-      if (botUser) {
+      if (!botUser) {
+        msg = await chan.send(embed.description, opts);
+      } else if (hook && !hasAttachment) { //Remove !hasAttachment when https://github.com/hydrabolt/discord.js/pull/1449 is fixed
+        if (hasAttachment) {
+          hookOpts.file = opts.file;
+          msg = await hook.send(embed.description, hookOpts);
```
Comment (Half-Shot) [inline src/bot.ts]: This should be the bot user, no?

File: src/bot.ts [hunk 2]
```diff
@@ -159,7 +159,11 @@ export class DiscordBot {
     const profile = result.botUser ? await mxClient.getProfileInfo(event.sender) : null;
     const embed = this.MatrixEventToEmbed(event, profile, chan);
     let opts : Discord.MessageOptions = {};
-    if (["m.image", "m.audio", "m.video", "m.file"].indexOf(event.content.msgtype) !== -1) {
+    let hookOpts : Discord.WebhookMessageOptions = {
+      username: event.sender,
```
Comment (Half-Shot) [inline src/bot.ts]: Embed will contain a display name and a avatar, so you could put this in here.

File: src/bot.ts (hunk 1 above)
Comment (Half-Shot) [inline src/bot.ts]: Looking at it, probably best to comment this out since the else aught to take care of this anyway

**Comment by Half-Shot** [comment]
LGTM

**Pull request merged**